UefiCpuPkg/Library/MmSaveStateLib/AmdMmSaveState.c

```diff
@@ -298,12 +305,5 @@ MmSaveStateGetRegisterLma (
   VOID
   )
 {
-  UINT32  LMAValue;
-
-  LMAValue = (UINT32)AsmReadMsr64 (EFER_ADDRESS) & LMA;
-  if (LMAValue) {
-    return EFI_MM_SAVE_STATE_REGISTER_LMA_64BIT;
-  }
-
-  return EFI_MM_SAVE_STATE_REGISTER_LMA_32BIT;
+  return EFI_MM_SAVE_STATE_REGISTER_LMA_64BIT;

```

Comment:
I think you keep the above check and ASSERT if LMAValue is not set.

Reply:
We (+abdattar) had a discussion internally as keeping the current code and adding a comment additionally.
> AMD64 processors support EFI_SMM_SAVE_STATE_REGISTER_LMA_64BIT only